src/components/ButtonState/ButtonState.test.ts

```diff
@@ -36,6 +36,24 @@ describe('template', () => {
 });
 
 describe('disabled', () => {
+    it('is false when not disabled', () => {
+        const wrapper = shallowMount(ButtonState, {props: {disabled: false}});
+
+        expect(wrapper.vm.disabled).toBe(false);
+    });
+
+    it('is true when disabled', () => {
+        const wrapper = shallowMount(ButtonState, {props: {disabled: true}});
+
+        expect(wrapper.vm.disabled).toBe(true);
+    });
+
+    it('is true when disabled and not loading', () => {
+        const wrapper = shallowMount(ButtonState, {props: {disabled: true, loading: false}});
+
+        expect(wrapper.vm.disabled).toBe(true);
+    });
```

Comment:
Feels like this is already tested by the previous test above.
I think 'is false when disabled and loading' is a better test.

Reply:
Wait, maybe it should also be disabled when loading haha